retrieve_address_book.go

```diff
@@ -0,0 +1,67 @@
+package hedera
+
+func GetMainnetAddressBook() (*NodeAddressBook, error) {
+	net := make(map[string]AccountID)
+	client := ClientForNetwork(net)
+
+	client.SetMirrorNetwork([]string{"mainnet-public.mirrornode.hedera.com:5600"})
+	ledger, err := LedgerIDFromNetworkName(NetworkNameMainnet)
+	if err != nil {
+		return &NodeAddressBook{}, err
+	}
+	client.SetLedgerID(*ledger)
+
+	result, err := NewAddressBookQuery().
+		SetFileID(FileID{0, 0, 101, nil}).
+		SetLimit(14).
+		Execute(client)
+	if err != nil {
+		return &NodeAddressBook{}, err
+	}
+
+	return &result, nil
+}
+
+func GetPreviewnetAddressBook() (*NodeAddressBook, error) {
+	net := make(map[string]AccountID)
+	client := ClientForNetwork(net)
+
+	client.SetMirrorNetwork([]string{"hcs.previewnet.mirrornode.hedera.com:5600"})
+	ledger, err := LedgerIDFromNetworkName(NetworkNamePreviewnet)
+	if err != nil {
+		return &NodeAddressBook{}, err
+	}
+	client.SetLedgerID(*ledger)
+
+	result, err := NewAddressBookQuery().
+		SetFileID(FileID{0, 0, 101, nil}).
+		SetLimit(14).
+		Execute(client)
+	if err != nil {
+		return &NodeAddressBook{}, err
+	}
+
+	return &result, nil
+}
+
+func GetTestnetAddressBook() (*NodeAddressBook, error) {
+	net := make(map[string]AccountID)
+	client := ClientForNetwork(net)
+
+	client.SetMirrorNetwork([]string{"hcs.testnet.mirrornode.hedera.com:5600"})
```

Comment:
This repeats the mirror node URLs from `client.go`. You should just use `ClientForPreviewnet`, etc.